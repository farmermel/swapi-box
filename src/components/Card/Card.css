.card {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  background-color: yellow;
  width: 180px;
  height: 300px;
  padding: 20px 10px;
  margin: 5px 40px; }

.card p {
  height: 30px; }

.bump-top,
.bump,
.bump-bottom {
  position: absolute;
  width: 10px;
  height: 40px;
  left: -10px;
  background-color: inherit; }

.bump-bottom {
  top: 250px; }

.bump-top {
  top: 50px; }
