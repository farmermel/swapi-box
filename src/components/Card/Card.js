import React from 'react';

const Card = ({ data }) => {
  return (
    <div>
      <h3>{ data.name }</h3>
      <p>{ data.species }</p>
      <p>{ data.homeworld }</p>
    </div>
  )
}

export default Card;