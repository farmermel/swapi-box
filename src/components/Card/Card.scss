.card {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  background-color: yellow;
  width: 180px;
  height: 300px;
  padding: 20px 10px;
  margin: 5px 40px;
}

.card p {
  height: 30px;
}

.bump-top,
.bump,
.bump-bottom {
  position: absolute;
  width: 10px;
  height: 40px;
  // z-index: 2;
  left: -10px;
  background-color: inherit;
  // background-color: green;
}

.bump-bottom {
  top: 250px;
}

.bump-top {
  top: 50px;
}