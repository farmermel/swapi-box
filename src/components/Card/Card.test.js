import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Card from './Card';
import mockData from '../../mock-data';

describe('Card', () => {
  it('matches the snapshot', () => {
    const cardData = mockData.cleaned.planet[0]
    let wrapper = shallow(<Card data={cardData}
                                toggleFav={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  })
})