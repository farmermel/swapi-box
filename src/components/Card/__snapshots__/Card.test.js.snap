// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "favorite": false,
            "id": 9823742,
            "info": Object {
              "climate": "hot",
              "name": "NABOO",
              "population": "4",
              "residents": "no residents",
              "terrain": "sandy",
            },
            "type": "planetData",
          }
    }
    toggleFav={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorites"
          id={undefined}
          onClick={[Function]}
/>,
        Array [
          <p>
            <span>
                        name
                        : 
            </span>
            NABOO
</p>,
          <p>
            <span>
                        terrain
                        : 
            </span>
            sandy
</p>,
          <p>
            <span>
                        population
                        : 
            </span>
            4
</p>,
          <p>
            <span>
                        climate
                        : 
            </span>
            hot
</p>,
          <p>
            <span>
                        residents
                        : 
            </span>
            no residents
</p>,
        ],
        <div
          className="bump-top"
/>,
        <div
          className="bump"
/>,
        <div
          className="bump-bottom"
/>,
      ],
      "className": "card planetData",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorites",
          "id": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "0NABOO",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              name
              : 
</span>,
            "NABOO",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "name",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "name",
              ": ",
            ],
            "type": "span",
          },
          "NABOO",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1sandy",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              terrain
              : 
</span>,
            "sandy",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "terrain",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "terrain",
              ": ",
            ],
            "type": "span",
          },
          "sandy",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "24",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              population
              : 
</span>,
            "4",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "population",
              ": ",
            ],
            "type": "span",
          },
          "4",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "3hot",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              climate
              : 
</span>,
            "hot",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "climate",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "climate",
              ": ",
            ],
            "type": "span",
          },
          "hot",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "4no residents",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              residents
              : 
</span>,
            "no residents",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "residents",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "residents",
              ": ",
            ],
            "type": "span",
          },
          "no residents",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bump-top",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bump",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bump-bottom",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorites"
            id={undefined}
            onClick={[Function]}
/>,
          Array [
            <p>
              <span>
                            name
                            : 
              </span>
              NABOO
</p>,
            <p>
              <span>
                            terrain
                            : 
              </span>
              sandy
</p>,
            <p>
              <span>
                            population
                            : 
              </span>
              4
</p>,
            <p>
              <span>
                            climate
                            : 
              </span>
              hot
</p>,
            <p>
              <span>
                            residents
                            : 
              </span>
              no residents
</p>,
          ],
          <div
            className="bump-top"
/>,
          <div
            className="bump"
/>,
          <div
            className="bump-bottom"
/>,
        ],
        "className": "card planetData",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorites",
            "id": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "0NABOO",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                name
                : 
</span>,
              "NABOO",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "name",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "name",
                ": ",
              ],
              "type": "span",
            },
            "NABOO",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1sandy",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                terrain
                : 
</span>,
              "sandy",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "terrain",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "terrain",
                ": ",
              ],
              "type": "span",
            },
            "sandy",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "24",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                population
                : 
</span>,
              "4",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "population",
                ": ",
              ],
              "type": "span",
            },
            "4",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "3hot",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                climate
                : 
</span>,
              "hot",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "climate",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "climate",
                ": ",
              ],
              "type": "span",
            },
            "hot",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "4no residents",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                residents
                : 
</span>,
              "no residents",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "residents",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "residents",
                ": ",
              ],
              "type": "span",
            },
            "no residents",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bump-top",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bump",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bump-bottom",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
